docs/deployment/docker-compose-fixes.md

```
@@ -0,0 +1,84 @@
+# Run with production config
+docker-compose -f docker-compose.yml -f docker-compose.prod.yml up
+```
+
+The configuration check script will validate all required variables before starting the application.
+
+## Best Practices
+
+1. **Never commit real secrets** - Always use environment variables or secret management systems
+2. **Use placeholder values only for builds** - The validation script ensures they're replaced before runtime
+3. **Document all required environment variables** - See `docs/deployment/production-environment-variables.md`
+4. **Use proper secret management** in production (e.g., HashiCorp Vault, AWS Secrets Manager, etc.)
```

markdownlint-cli2 (0.17.2)

Files should end with a single newline character

(MD047, single-trailing-newline)